src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts

```diff
@@ -196,7 +196,13 @@ async function webviewPreloads(ctx: PreloadContext) {
 	// check if an input element is focused within the output element
 	const checkOutputInputFocus = (e: FocusEvent) => {
 		lastFocusedOutput = getOutputContainer(e);
-		const activeElement = window.document.activeElement;
+		let activeElement = window.document.activeElement;
+
+		// Recursively check for shadow root descendants
+		while (activeElement?.shadowRoot) {
```

Comment:
don't just dig to the inner most element, check at each point if the element is editable to follow the snippet example